python.md

```diff
@@ -0,0 +1,216 @@
+[Indentation](https://google.github.io/styleguide/pyguide.html#34-indentation)]*
+
+Use **4 spaces** per indentation level for new projects, but favor consistency
+in existing projects (most of them use 2 spaces).
+
+Favor **aligned indentation** over hanging indentation for line continuation
+and resolve potential problems like so:
+
+- **Problem:** Long first lines push continued lines to the right edge:
+
+  ```python
+  # NO:
+  import package.has.many.sub.packages.or_long_package_names.foo
+
+  # This problem occurs when picking long function names ...
+  def this_function_has_a_very_long_name_and_many_arguments(lorem, ipsum_dolor,
+                                                            sit, amet,
+                                                            consectetur):
+    # ... or as result of 'import <full module path>' syntax.
+    package.has.many.sub.packages.or_long_package_names.foo.bar(lorem,
+                                                                ipsum_dolor,
+                                                                sit, amet,
+                                                                consectetur)
+  ```
+
+  **Solution:** Avoid long function names and use `from <full package path>
+  import <module>`- syntax:
+
+  ```python
+  # YES:
+  from package.has.many.sub.packages.or_long_package_names import foo
+
+  def call_foo_bar(lorem, ipsum_dolor, sit, amet, consectetur):
+    foo.bar(lorem, ipsum_dolor, sit, amet, consectetur)
+  ```
+
+- **Problem:** Indentation of opening parenthesis might coincide with logical
+  indentation of nested block:
+
+  ```python
+  # NO:
+  # As a matter of fact, it will always coincide for 'if' statements
+  if (lorem and ipsum_dolor and sit and amet and consectetur and
+      adipisicing and elit):
+      foo()
+  ```
+
+  **Solution:** Start nested block with a comment to add visual distinction:
+
+  ```python
+  # YES:
+  if (lorem and ipsum_dolor and sit and amet and consectetur and
+      adipisicing and elit):
+      # TODO: Add meaningful comment about 'foo()' and its conditions
+      foo()
+  ```
+  Note that function definitions may also deindent the closing parenthesis to
+  align with the `def` keyword as demonstrated in the Google Python Style Guide
+  [Type
+  Annotations](https://google.github.io/styleguide/pyguide.html#3192-line-breaking)
+  section.
+
+- **Problem:** Renames in first line might require re-aligning continued lines.\
+  **Solution:** Avoid renames by picking [good
+  names](https://duckduckgo.com/?q=There+are+only+two+hard+things+in+Computer+Science%3A+cache+invalidation+and+naming+things)
+  right away. :P
+
+- **Special case:** Also align continued lines of continued lines:
+
+  ```python
+
+  # YES:
+  raise Exception("Lorem {0} ipsum {1} dolor sit amet, {2} consectetur {3} ad "
+                  "{4} rcitatio {5}.".format(reprehenderit, voluptate, velit,
+                                             pariatur, deserunt, laborum))
```

Comment:
I support the idea to align the continued lines with the opening bracket, but this seems this is the opposite of Google's line breaking rules, right?
https://google.github.io/styleguide/pyguide.html#3192-line-breaking

Just wanted to make sure I understand it correctly. :)